File: cpp/219.cpp
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
  bool containsNearbyDuplicate(vector<int> &nums, int k) {
    unordered_map<int, int> m;
    for (int i = 0; i < nums.size(); i++) {
      if (i - k - 1 >= 0)
        --m[nums[i - k - 1]];
      if (m[nums[i]])
        return true;
      ++m[nums[i]];
    }
    return false;
  }
};

File: python/219.py
import collections


class Solution(object):
    def containsNearbyDuplicate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: bool
        """
        c = collections.Counter()
        for i, n in enumerate(nums):
            if i-k-1 >= 0:
                c[nums[i-k-1]] -= 1
            if c[n]:
                return True
            c[n] += 1
        return False
